File: Top interview 150/Array and string/380.java
import java.util.HashSet;

class RandomizedSet {
    private HashSet<Integer> set;

    public RandomizedSet() {
        set = new HashSet<>();
    }

    public boolean insert(int val) {
        boolean res = set.contains(val);
        if (res)
            return false;
        set.add(val);
        return true;
    }

    public boolean remove(int val) {
        boolean res = set.contains(val);
        if (!res)
            return false;
        set.remove(val);
        return true;
    }

    public int getRandom() {
        int random = (int) (Math.random() * set.size());
        var iterator = set.iterator();
        for (int i = 0; i < random - 1; i++) {
            iterator.next();
        }
        return iterator.next();
    }
}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * RandomizedSet obj = new RandomizedSet();
 * boolean param_1 = obj.insert(val);
 * boolean param_2 = obj.remove(val);
 * int param_3 = obj.getRandom();
 */

File: daily/January/380.cs
namespace RandomizedSet;
public class RandomizedSet
{
    private Dictionary<int, int> dict;
    private List<int> list;
    private Random rand;

    public RandomizedSet()
    {
        dict = new Dictionary<int, int>();
        list = new List<int>();
        rand = new Random();
    }

    public bool Insert(int val)
    {
        if (dict.ContainsKey(val))
        {
            return false;
        }
        dict.Add(val, list.Count);
        list.Add(val);
        return true;
    }

    public bool Remove(int val)
    {
        if (!dict.ContainsKey(val))
        {
            return false;
        }
        int lastElement = list[list.Count - 1];
        int index = dict[val];
        list[index] = lastElement;
        dict[lastElement] = index;
        list.RemoveAt(list.Count - 1);
        dict.Remove(val);
        return true;
    }

    public int GetRandom()
    {
        return list[rand.Next(list.Count)];
    }
}